specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2-preview/rbac.json

```
@@ -551,31 +550,395 @@
+              "modelAsString": true,
+              "values": [
+                {
+                  "name": "ReadHsmKey",
+                  "value": "Microsoft.KeyVault/managedHsm/keys/read/action",
+                  "description": "Read HSM key."
+                },
+                {
+                  "name": "WriteHsmKey",
+                  "value": "Microsoft.KeyVault/managedHsm/keys/write/action",
+                  "description": "Create or update HSM key."
+                },
+                {
+                  "name": "ReadDeletedHsmKey",
+                  "value": "Microsoft.KeyVault/managedHsm/keys/deletedKeys/read/action",
+                  "description": "Read deleted HSM key."
+                },
+                {
+                  "name": "RecoverDeletedHsmKey",
+                  "value": "Microsoft.KeyVault/managedHsm/keys/deletedKeys/recover/action",
+                  "description": "Recover deleted HSM key."
+                },
+                {
+                  "name": "BackupHsmKeys",
+                  "value": "Microsoft.KeyVault/managedHsm/keys/backup/action",
+                  "description": "Backup HSM keys."
+                },
+                {
+                  "name": "RestoreHsmKey",
+                  "value": "Microsoft.KeyVault/managedHsm/keys/restore/action",
+                  "description": "Restore HSM keys."
+                },
+                {
+                  "name": "DeleteRoleAssignment",
+                  "value": "Microsoft.KeyVault/managedHsm/roleAssignments/delete/action",
+                  "description": "Delete role assignment."
+                },
+                {
+                  "name": "GetRoleRoleAssignment",
+                  "value": "Microsoft.KeyVault/managedHsm/roleAssignments/read/action",
+                  "description": "Get role assignment."
+                },
+                {
+                  "name": "WriteRoleAssignment",
+                  "value": "Microsoft.KeyVault/managedHsm/roleAssignments/write/action",
+                  "description": "Create or update role assignment."
+                },
+                {
+                  "name": "ReadRoleDefinition",
+                  "value": "Microsoft.KeyVault/managedHsm/roleDefinitions/read/action",
+                  "description": "Get role definition."
+                },
+                {
+                  "name": "EncryptHsmKeys",
+                  "value": "Microsoft.KeyVault/managedHsm/keys/encrypt/action",
+                  "description": "Encrypt HSM keys."
+                },
+                {
+                  "name": "DecryptHsmKeys",
+                  "value": "Microsoft.KeyVault/managedHsm/keys/decrypt/action",
+                  "description": "Decrypt HSM keys."
+                },
+                {
+                  "name": "WrapHsmKeys",
+                  "value": "Microsoft.KeyVault/managedHsm/keys/wrap/action",
+                  "description": "Wrap HSM keys."
```

Nit (it's been singular up till now, and technically you're only wrapping a key - those the scope is keys; maybe just be consistent throughout):
```suggestion
                  "description": "Wrap an HSM key."
```

Same consideration below (I'll not comment on all of them).

---

this is actually "Wrap *with* an HSM key"